source/lac/trilinos_tpetra_vector.cc

```diff
@@ -0,0 +1,686 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/std_cxx14/memory.h>
+
+#include <deal.II/lac/trilinos_tpetra_vector.h>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  ifdef DEAL_II_WITH_MPI
+
+#    include <deal.II/base/index_set.h>
+
+#    include <deal.II/lac/read_write_vector.h>
+
+#    include <boost/io/ios_state.hpp>
+
+#    include <Teuchos_DefaultMpiComm.hpp>
+#    include <Tpetra_Import_def.hpp>
+#    include <Tpetra_Map_def.hpp>
+
+#    include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  namespace TpetraWrappers
+  {
+    template <typename Number>
+    Vector<Number>::Vector()
+      : vector(new Tpetra::Vector<Number, int, types::global_dof_index>(
+          Teuchos::RCP<Tpetra::Map<int, types::global_dof_index>>(
+            new Tpetra::Map<int, types::global_dof_index>(
+              0,
+              0,
+              Utilities::Trilinos::tpetra_comm_self()))))
+    {}
+
+
+
+    template <typename Number>
+    Vector<Number>::Vector(const Vector<Number> &V)
+      : Subscriptor()
```

Comment:
Why is it the only constructor where you call `Subscriptor()` ?

Reply:
It really doesn't matter which constructor we are calling, they all do the same. I'll add the base class constructor call to the other constructors as well (and do the same for `EptraWrappers`).